### ISSUE TITLE
unreachable code when creating variable

### ISSUE THREAD

**Issue by aeosynth** [nushell/nushell#3350]
```
> let a = (echo $b | c)
thread 'main' panicked at 'internal error: entered unreachable code', crates/nu-engine/src/evaluate/evaluator.rs:62:22
```

```
> version | pivot
────┬────────────────────┬─────────────────────────────────────────────
 #  │      Column0       │                   Column1
────┼────────────────────┼─────────────────────────────────────────────
  0 │ version            │ 0.30.0
  1 │ short_commit       │ 02d57299
  2 │ commit_hash        │ 02d5729941c9e5ad013889e1b9c452cc40a8e4df
  3 │ commit_date        │ 2021-04-21 02:31:54
  4 │ build_os           │ linux-x86_64
  5 │ rust_version       │ rustc 1.51.0 (2fd73fabe 2021-03-23)
  6 │ rust_channel       │ stable-x86_64-unknown-linux-gnu (default)
  7 │ cargo_version      │ cargo 1.51.0 (43b129a20 2021-03-16)
  8 │ pkg_version        │ 0.30.0
  9 │ build_time         │ 2021-04-21 05:27:45
 10 │ build_rust_channel │ release
 11 │ features           │ clipboard-cli, ctrlc, default,
    │                    │ directories, dirs, ichwh, ptree, rustyline,
    │                    │ term, trash, uuid, which, zip

```

**Comment by fdncred** [comment]
I wonder if this is because your syntax is wrong above. It should be `let a = $(echo $b | c)`. Granted, we probably shouldn't panic anyway.

**Comment by jonathandturner** [comment]
Just tried this out and this is fixed by #3389. Wanted to note that so we didn't accidentally try to fix this twice.
